### PR TITLE
Adds builder onboarding

### LINKED ISSUE #8887: Implement builder onboarding

### PR DESCRIPTION
**Pull request by javierarce**
This PR: 
- Adds the Builder onboarding to explain the basic UI concepts of the new design.
- Uses a permanent storage for the notifications introduced with #9220  (https://github.com/CartoDB/cartodb/pull/9238/commits/d23adddf85333bed5313829b4d0dd266b5958a54, https://github.com/CartoDB/cartodb/pull/9238/commits/1a040f73018481b70dbec4df854966f7cc112e3e)

Closes #8887

### CODE REVIEW

**Comment by javierarce** [comment]
CR: @xavijam

File: app/assets/stylesheets/common/builder-onboarding.css.scss [hunk 1]
```diff
@@ -0,0 +1,216 @@
+@import '../variables/mixins';
```
Comment (xavijam) [inline app/assets/stylesheets/common/builder-onboarding.css.scss]: Do we provide formula with strings? ^^^ I'd change at least the fake formula widget :)

**Comment by xavijam** [comment]
Could you help me with styles @piensaenpixel?

File: lib/assets/javascripts/cartodb3/components/onboardings/analysis/analysis-view.js [hunk 2]
```diff
@@ -9,7 +9,6 @@ module.exports = CoreView.extend({
   className: 'AnalysisCompletionDetails is-opening',
 
   events: {
-    'click .js-forget': '_onClickForget',
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/onboardings/analysis/analysis-view.js]: no forget anymore?

File: lib/assets/javascripts/cartodb3/components/onboardings/builder/builder-launcher.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    init: function (opts) {
+      init(opts);
+
+      LocalStorage.init(STORAGE_KEY, {
+        userModel: userModel
+      });
+    },
+
+    launch: function (type, model) {
+      if (!LocalStorage.get(type) && ANALYSES_TYPES[type]) {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/onboardings/builder/builder-launcher.js]: LocalStorage for analysis and for builder onboardings?

File: lib/assets/javascripts/cartodb3/components/onboardings/builder/builder-view.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    this._modalModel = opts.modalModel;
+    this._userModel = opts.userModel;
+    this._userNotifications = opts.userNotifications;
+    this.model = new cdb.core.Model({
+      step: 0
+    });
+
+    this.model.bind('change:step', this._onChangeStep, this);
+
+    $(document).on('keydown', this._onKeyDown.bind(this));
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/onboardings/builder/builder-view.js]: I miss the unbind of this binding.

File: lib/assets/javascripts/cartodb3/components/onboardings/builder/builder-launcher.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
+var LocalStorage = require('../../../components/local-storage/local-storage');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/onboardings/builder/builder-launcher.js]: analysis-launcher?

File: lib/assets/javascripts/cartodb3/data/user-notifications.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ */
+
+var UserNotifications = Backbone.Model.extend({
+  sync: function (method, model, options) {
+    return Backbone.sync('update', model, options);
+  },
+
+  url: function () {
+    return '/api/v3/notifications/' + this.get('key');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/data/user-notifications.js]: This url needs the `baseUrl` from the `configModel`.

**Comment by xavijam** [comment]
Why are there two different launchers?

File: lib/assets/javascripts/cartodb3/components/onboardings/analysis/analysis-view.js (hunk 2 above)
Comment (javierarce) [inline lib/assets/javascripts/cartodb3/components/onboardings/analysis/analysis-view.js]: Not really needed, I check the status of the checkbox.

File: lib/assets/javascripts/cartodb3/components/onboardings/builder/builder-launcher.js (hunk 5 above)
Comment (javierarce) [inline lib/assets/javascripts/cartodb3/components/onboardings/builder/builder-launcher.js]: Sorry, this file is not actually used. I created at the beginning and forgot to remove it. I'll do it now.

File: lib/assets/javascripts/cartodb3/data/user-notifications.js (hunk 6 above)
Comment (javierarce) [inline lib/assets/javascripts/cartodb3/data/user-notifications.js]: Totally right. Fixed.

File: lib/assets/javascripts/cartodb3/components/onboardings/builder/builder-launcher.js (hunk 3 above)
Comment (javierarce) [inline lib/assets/javascripts/cartodb3/components/onboardings/builder/builder-launcher.js]: Sorry, this file is not currently used. Forgot to remove it.

File: lib/assets/javascripts/cartodb3/components/onboardings/builder/builder-view.js (hunk 4 above)
Comment (javierarce) [inline lib/assets/javascripts/cartodb3/components/onboardings/builder/builder-view.js]: Added.

Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/onboardings/builder/builder-view.js]: this._onKeyDown.bind(this) inside the on will give us problems, already suffered. What we are doing lately is:

```js
this._keyDown = this._keyDown.bind(this);

$(document).on('keydown', this._onKeyDown);
…

$(document).off('keydown', this._onKeyDown);
```

Comment (javierarce) [inline lib/assets/javascripts/cartodb3/components/onboardings/builder/builder-view.js]: I didn't encounter any problem, but I've made the change you suggested. Thanks!

File: app/assets/stylesheets/common/builder-onboarding.css.scss [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+.BuilderOnboarding-pad4,
+.BuilderOnboarding-pad5 {
+  @include transition(all, 200ms, ease-in-out, 300ms);
+  position: absolute;
+  top: 0;
+  right: -$sWidgetsPadWidth;
+  width: $sWidgetsPadWidth;
+  height: 100%;
+  background: image-url('layout/widgets.png') no-repeat center center;
+  background-color: #F2F6F9;
```
Comment (piensaenpixel) [inline app/assets/stylesheets/common/builder-onboarding.css.scss]: $cSecondaryBackground ?

**Comment by piensaenpixel** [comment]
very minor comment, the rest LGTM 💃

File: app/assets/stylesheets/common/builder-onboarding.css.scss (hunk 7 above)
Comment (javierarce) [inline app/assets/stylesheets/common/builder-onboarding.css.scss]: Thanks!

**Pull request merged**